meson.build

```diff
@@ -0,0 +1,305 @@
+  '-Wno-unused-parameter',
+  '-Wunknown-attributes',
+  '-Wtypedef-redefinition',
+  '-Warray-bounds',
+  '-Wparentheses-equality',
+  '-Wunused-value',
+  '-Wmissing-braces',
+]
+cc = meson.get_compiler('c')
+foreach arg: test_args
+  if cc.has_argument(arg)
+    add_project_arguments(arg, language : 'c')
+  endif
+endforeach
+
+if get_option('enable-more-warnings') == 'yes' or get_option('enable-more-warnings') == 'error'
+  if cc.has_argument('-Wextra')
+    add_project_arguments('-Wextra', language : 'c')
+  endif
+
+  if cc.has_argument('-Wall')
+    add_project_arguments('-Wall', language : 'c')
+  endif
+
+  if get_option('enable-more-warnings') == 'error' and cc.has_argument('-Werror')
+    add_project_arguments('-Werror', language : 'c')
+  endif
+endif
+
+if get_option('enable-lto')
+  if cc.has_argument('-flto')
+    add_project_arguments('-flto', language : 'c')
+  endif
+endif
+
+if get_option('enable-ld-gc')
+  link_gc = '-Wl,--gc-sections'
+  if cc.has_argument('-fdata-sections') and cc.has_argument('-ffunction-sections') and cc.links('int main () { return 0; }', name: link_gc, args: link_gc)
+    add_project_arguments('-fdata-sections', language : 'c')
+    add_project_arguments('-ffunction-sections', language : 'c')
+    add_project_arguments('-Wl,--gc-sections', language : 'c')
+  else
+    error('Disable garbage collector due to unsupported compiler flags.')
+  endif
+endif
+
+gnome = import('gnome')
+
+asresources = gnome.compile_resources(
+  'nm-anyconnect-dialog.ui', 'properties/gresource.xml',
+  source_dir : 'properties',
+  c_name : 'as')
+
+# glib-compile-resources
+glib_resources_path = join_paths(meson.build_root(), 'properties/')
+r = run_command('mkdir', glib_resources_path)
+if r.returncode() != 0
+  message('Could not compile glib resources:')
+  message(r.stderr().strip())
+endif
+
+r = run_command('glib-compile-resources', '--target', join_paths(glib_resources_path, 'resources.h'), '--sourcedir=properties/', '--generate-header', 'properties/gresource.xml')
+if r.returncode() != 0
+  message('Could not compile glib resources:')
+  error(r.stderr().strip())
+endif
+
+r = run_command('glib-compile-resources', '--target', join_paths(glib_resources_path, 'resources.c'), '--sourcedir=properties/', '--generate-source', 'properties/gresource.xml')
+if r.returncode() != 0
+  message('Could not compile glib resources:')
+  error(r.stderr().strip())
+endif
+
+# Generating appdata
+service_name = custom_target ('nm-anyconnect-service-name',
+                              output: 'nm-anyconnect-service.name',
+                              input: 'nm-anyconnect-service.name.in',
+                              command: ['sed', '-e', 's|[@]LIBEXECDIR[@]|' + libexecdir + '|g', '-e', 's|[@]PLUGINDIR[@]/|' + plugindir + '|g', '@INPUT@'],
+                              capture: true,
+                              install: true,
+                              install_dir: join_paths(prefix, 'lib', 'NetworkManager' , 'VPN/'))
```

Comment:
I think `.in` files are supposed to be processed using the `configuration_file` directive, see https://mesonbuild.com/Configuration.html